### ISSUE TITLE
pyright thinks the code is unreachable

### ISSUE THREAD

**Issue by Y7n05h** [microsoft/pyright#2286]
**Describe the bug**
I found that pyright always thinks that the code after `send()` in the function is unreachable.
But in fact, this script has no unreachable statements.

```Python
import pwn
io = pwn.process(["/usr/bin/cat", "/dev/stdin"])


def Test():
    io.send(b"home")
    print("pyright thinks this statement is unreachable")


Test()
```

**To Reproduce**
install [pwntools](https://github.com/Gallopsled/pwntools)  and open the file

**Expected behavior**

Optimize the way to detect whether the code is reachable

**Screenshots or Code**
![2021-09-10_18-30](https://user-images.githubusercontent.com/69407218/132840828-8473f4b7-5272-4c7a-87ca-8751b0e9b510.png)

**VS Code extension or command-line**
coc-pyright 1.1.166 with coc.nvim

**Comment by erictraut** [comment]
Where is the type information for the `pwn` library coming from? It doesn't appear to be a "py.typed" library, and it doesn't have any inlined types. I'm guessing that you have enabled the "python.analysis.useLibraryCodeForTypes" setting? This setting tells pyright to try to infer types from (possibly untyped) Python library code. Type inference of this sort is error prone. It's good enough for producing reasonable completion suggestions, but it will typically result in incorrect behavior if you have enabled static type checking, so we don't recommend enabling "useLibraryCodeForTypes" if you enable type checking.

In this particular case, the type analyzer is inferring the return type of `io.send`. That method calls `self.send_raw(data)`, which is implemented with the statement `raise EOFError('Not Implemented')`. The method `send_raw` is not marked as abstract, and it doesn't raise a `NotImplementedError`, so pyright assumes that it always raises an exception and is not overridden.

The best solution here is to use high-quality type stubs for this library or (even better) convince the maintainers to add inlined type information for it. As a smaller change, you could get them to either mark the `send_raw` method with an `@abstractmethod` decorator or switch to `NotImplementedError` exception. Disabling "useLibraryCodeForTypes" should also fix the problem because pyright won't attempt to infer the types of any symbols imported from untyped libraries.
